### PR TITLE
Run benchmark tests only during merges to develop

### PR DESCRIPTION
**Pull request by praveenrewar**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by joaopapereira — APPROVED** [review]
LGTM

**Comment by neil-hickey** [comment]
@praveenrewar did we end up deciding this was useful or not? Happy to merge if it is, or else we can close this out

**Comment by praveenrewar** [comment]
IIRC, we decided that we don't need this, but in future, instead of not running the workflow I think we should run the test for merge requests to ensure that we get the results before merging a PR and not afterwards.

**Pull request closed without merging**
